src/test/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheImplTest.java

```diff
@@ -0,0 +1,92 @@
+package co.rsk.federate.btcreleaseclient.cache;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import co.rsk.crypto.Keccak256;
+import co.rsk.federate.signing.utils.TestUtils;
+import java.time.Duration;
+import java.time.Instant;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class PegoutSignedCacheImplTest {
+
+  private static final Duration DEFAULT_TTL = Duration.ofMinutes(30);
+  private static final Keccak256 PEGOUT_CREATION_RSK_HASH = TestUtils.createHash(1);
+
+  private final PegoutSignedCacheImpl pegoutSignedCache = new PegoutSignedCacheImpl(DEFAULT_TTL);
+
+  @BeforeEach
+  void setUp() {
+    pegoutSignedCache.getCache().clear();
+  }
+
+  @Test
+  void hasAlreadyBeenSigned_shouldReturnFalse_whenPegoutCreationRskTxHashIsNull() {
+    Keccak256 pegoutCreationRskTxHash = null;
+
+    boolean result = pegoutSignedCache.hasAlreadyBeenSigned(pegoutCreationRskTxHash);
+
+    assertFalse(result);
+  }
+
+  @Test
+  void hasAlreadyBeenSigned_shouldReturnFalse_whenPegoutCreationRskTxHashIsNotCached() {
+    boolean result = pegoutSignedCache.hasAlreadyBeenSigned(PEGOUT_CREATION_RSK_HASH);
+
+    assertFalse(result);
+  }
+
+  @Test
+  void hasAlreadyBeenSigned_shouldReturnFalse_whenCacheContainsInvalidTimestamp() {
+    Instant currentTimestamp = Instant.now();
+    Instant timestamp = currentTimestamp.minusMillis(Duration.ofMinutes(60).toMillis());
+    pegoutSignedCache.getCache().put(PEGOUT_CREATION_RSK_HASH, timestamp);
+
+    boolean result = pegoutSignedCache.hasAlreadyBeenSigned(PEGOUT_CREATION_RSK_HASH);
+
+    assertFalse(result);
+  }
+
+  @Test
+  void hasAlreadyBeenSigned_shouldReturnTrue_whenCacheContainsValidTimestamp() {
+    Instant currentTimestamp = Instant.now();
+    Instant timestamp = currentTimestamp.minusMillis(Duration.ofMinutes(10).toMillis());
+    pegoutSignedCache.getCache().put(PEGOUT_CREATION_RSK_HASH, timestamp);
+
+    boolean result = pegoutSignedCache.hasAlreadyBeenSigned(PEGOUT_CREATION_RSK_HASH);
+
+    assertTrue(result);
+  }
+
+  @Test
+  void put_shouldNotPutInCache_whenPegoutCreationRskTxHashIsNull() {
+    Keccak256 pegoutCreationRskTxHash = null;
+
+    pegoutSignedCache.put(pegoutCreationRskTxHash);
```

Comment:
Should this throw an exception? It could be deceiving if somebody tries to insert a `null` value, no error is thrown but also nothing is saved. No way of knowing that something isn't right

Reply:
What exception should we throw in this case?

Reply:
IllegalArgumentException? I think it's better thp throw an exception than to simply ignore the fact that somebody tried to insert a null hash. That could mean something is wrong

Reply:
I agree. I would rather throw an NPE.

From the javadoc of the cache `putIfAbsent` method:

> if the specified key or value is null, and this map does not permit null keys or values

I think we can use this exception since that is what the underlying cache is doing. Anyways in the meantime I have added the exception and updated unit test. Wdyt?